src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1112,6 +1123,148 @@ protected static ImmutableArray<string> CheckTupleElementNames(int cardinality,
         public bool HasImplicitConversion(ITypeSymbol fromType, ITypeSymbol toType)
             => fromType != null && toType != null && this.ClassifyCommonConversion(fromType, toType).IsImplicit;
 
+        /// <summary>
+        /// Checks if <paramref name="symbol"/> is accessible from within <paramref name="within"/>. An optional qualifier of type
+        /// <paramref name="throughType"/> is used to resolve protected access for instance members. All symbols are
+        /// required to be from this compilation or some assembly referenced (<see cref="References"/>) by this
+        /// compilation. <paramref name="within"/> is required to be an <see cref="ITypeSymbol"/> or <see cref="IAssemblySymbol"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>Submissions can reference symbols from previous submissions and their referenced assemblies, even
+        /// though those references are missing from <see cref="References"/>.
+        /// See https://github.com/dotnet/roslyn/issues/27356.
+        /// This implementation works around that by permitting symbols from previous submissions as well.</para>
+        /// <para>It is advised to avoid the use of this API within the compilers, as the compilers have additional
```

Comment:
>It is advised to avoid the use of this API within the compilers [](start = 18, length = 63)

Would it be possible to put some enforcement in place. Perhaps in the form of an obsolete API in derived classes?

Reply:
Done.

---
In reply to: [192965273](https://github.com/dotnet/roslyn/pull/27312#discussion_r192965273) [](ancestors = 192965273)